redis/src/main/scala/zio/redis/Output.scala

```diff
@@ -401,6 +401,48 @@ object Output {
     }
   }
 
+  case object KeyValueOutput extends Output[Map[String, String]] {
+    protected def tryDecode(text: String) =
+      if (text.startsWith("*"))
+        parse(text)
+      else
+        throw ProtocolError(s"$text isn't a string.")
+
+    private[this] def parse(text: String): Map[String, String] = {
+      val data   = unsafeReadChunk(text, 0)
+      var output = Map.empty[String, String]
```

Comment:
Replace with mutable structure instead. You can convert to immutable map in the end.